docs/saf-authorized-endpoints.md

```diff
@@ -0,0 +1,101 @@
+
+# Secured endpoints with SAF resource 
+
+Any endpoint can have limited access by checking SAF resource of logged user. This feature is implemented in module
+`apiml-security-common`. Therefore, it can be used in each application with this dependency.
+
+Verification of SAF resource can be provided via three providers:
+- _the highest priority_: REST endpoint call (ZSS or similar one)
+- JZOS
+- _the lowest priority_: dummy implementation (defined in a file)
+
+**note**: The first available by priority will be used.
+
+## Checking providers
+
+### REST endpoint call
+
+This provider is once way how to enable the feature outside the mainframe (ie. running in Docker).
+
+- Method: `GET`
+- URL: `{base path}/{userId}/{class}/{entity}/{level}`
+- Response:
+```json5
+    {
+        "authorized": "{true|false}",
+        "error": "{true|false}",
+        "message": "{message}"
+    }
+```
+
+**note**: see also ZSS implementation https://github.com/zowe/zss/blob/master/c/authService.c
+
+#### Configuration
+
+`apiml.security.safEndpoint.enabled`
+- `true` or `false`, 'false' as default
+- to enable endpoint
+
+`apiml.security.safEndpoint.url`
+- base path of endpoint's URL (`{base path}/{userId}/{class}/{entity}/{level}`) 
+- as default `http://localhost:8542/saf-auth` (default location of ZSS endpoint)
+
+### JZOS
+
+This provider is the easiest way, how to use the feature on the mainframe.
+
+Enable, when classes `com.ibm.os390.security.PlatformAccessControl` and `com.ibm.os390.security.PlatformReturned`
+are available on classpath. It uses [method](https://www.ibm.com/support/knowledgecenter/SSYKE2_8.0.0/com.ibm.java.zsecurity.api.80.doc/com.ibm.os390.security/com/ibm/os390/security/PlatformAccessControl.html?view=kc#checkPermission-java.lang.String-java.lang.String-java.lang.String-int-), so
+for right using, definition of method must be also matching.
+
+### Dummy implementation
+
+This provider is for testing purpose outside the mainframe.
+
+You can create file `saf.yml` and locate it in the folder, where is application running or create file `mock-saf.yml` in the
+test module (root folder). The highest priority is to read file outside the JAR. A file (inner or outside) has to exist.
+
+Structure of YML file:
+```yaml
+  safAccess:
+    {CLASS}:
+      {RESOURCE}:
+        - {UserID}
+```
+
+**notes**:
+- Classes and resources are mapped into a map, user IDs into list.
+- load method does not support formatting with dots, like {CLASS}.{RESOURCE}, each element has to be separated
+- safAccess is not required to define empty file
+- classes and resources cannot be defined without user ID list
+- method could be called multiple times, definition are added to current (loaded)
+- when a user has multiple definition of same class and resource, just the highest access level is loaded
```

Comment:
By highest you mean the first defined in the order of reading of the files?

Reply:
No, it is by meaning, ie. UPDATE > READ